### ISSUE TITLE
gbm predict failure under caret

### ISSUE THREAD

**Issue by spedygiorgio** [topepo/caret#330]
I use GBM models and CARET R package routinously for my predictive modeling task. I seldom encounter an issue with joint use of GBM and CARET. GBM models appears to be properly tuned (in sample performance measures under cross - validation are returned). When I try to apply the model in prediction the following error is raised: "Error in object$var.levels[[i]] :  subscript out of bounds". I think this problem is shared by more people, considering the popularity of the GBM models.
I could not fix this interacting with package's authors. A possible hint might come from : https://groups.google.com/forum/#!topic/gbm-dev/y4XMyU2S3iQ.

Also, you can find data and code for a reproducible example (I hope) here: https://www.dropbox.com/sh/zor2ilvmd3ll6m9/AAAIShqwJ62YBKMyJZIb3Mzra?dl=0

**Comment by zachmayer** [comment]
It looks like you have a factor level in your test set that isn't in the training set, but I'm not sure.




Please provided a MINIMAL reproducible example, as code in a comment here on GitHub:




http://stackoverflow.com/a/5963610




—

**Comment by topepo** [comment]
You should subset the predictor columns as you did when calling `train`:

```r
dataTrain$predGbm <- predict(model.gbm,newdata=dataTrain[,predictorsSet],type="prob")["Yes"][,1]
```

**Comment by spedygiorgio** [comment]
Thank you really very much, Max. It works now. It is a very common mistake when using gbm + caret. Many people are not aware of that. It is possible to automate subset of predictors columns in the next release of caret? Regards
